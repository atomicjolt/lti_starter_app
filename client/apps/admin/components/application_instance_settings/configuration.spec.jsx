import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import Configuration from './configuration';

const mockStore = configureStore([]);

const store = mockStore({
  applicationInstances : {},
});

describe('configuration', () => {

  const params = {};
  it('renders the DownloadButton component', () => {
    const result = TestRenderer.create(
      <Provider store={store}>
        <Configuration params={params}/>
      </Provider>
    );
    expect(result).toMatchSnapshot();
  });
});
