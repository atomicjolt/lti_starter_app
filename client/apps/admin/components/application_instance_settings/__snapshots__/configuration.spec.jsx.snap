// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configuration renders the DownloadButton component 1`] = `
<div>
  <div
    className="o-grid__item u-full"
  >
    <label
      className="c-input c-input--dark"
      htmlFor="application_instance_config"
    >
      <span>
        Custom Application Instance Configuration
      </span>
      <textarea
        id="application_instance_config"
        name="config"
        onChange={[Function]}
        placeholder="ex: { \\"foo\\": \\"bar\\" }"
        value="{}"
      />
    </label>
  </div>
  <div
    className="o-grid__item u-full"
  >
    <label
      className="c-input c-input--dark"
      htmlFor="application_instance_lti_config"
    >
      <span>
        LTI Configuration
      </span>
      <textarea
        id="application_instance_lti_config"
        name="lti_config"
        onChange={[Function]}
        rows={25}
        value="{}"
      />
    </label>
  </div>
</div>
`;
